hack/e2e-test/run-e2e-test.sh

```diff
@@ -0,0 +1,422 @@
+  else
+    az storage container create --account-name "${STORAGE_ACCOUNT}" --account-key "${STORAGE_KEY}" --name "${TEST_ID}"
+  fi
+  echo "Successfully created test bucket."
+  echo "Setting up Azure infrastructure completed."
+}
+
+function setup_azcli() {
+  if ! $(which az > /dev/null); then
+    echo "Installing azure-cli"
+    apt update
+    apt install -y curl
+    curl -sL https://aka.ms/InstallAzureCLIDeb | bash
+    echo "Successfully installed azure-cli."
+  else
+    echo "azure-cli is already installed."
+  fi
+}
+
+function usage_azure() {
+  cat <<EOM
+Usage:
+
+Please make sure the following environment variables are set:
+
+    STORAGE_ACCOUNT                   Name of the storage account.
+    STORAGE_KEY                       Key of the storage account.
+    AZURE_STORAGE_API_ENDPOINT        URL of the Azure storage endpoint. ( required only for testing with Azurite)
+    AZURE_EMULATOR_ENABLED            Set to "true" to Enable the Azure emulator for testing. ( required only for testing with Azurite)
+    AZURITE_HOST                      Host of the Azurite service. ( required only for testing with Azurite)
+    AZURE_STORAGE_CONNECTION_STRING   Connection string for the Azure storage account. ( required only for testing with Azurite)
+EOM
+  exit 1
+}
+
+function setup_azure_e2e() {
+  if [[ -z ${STORAGE_ACCOUNT:-""} ]] || [[ -z ${STORAGE_KEY:-""} ]]; then
+      usage_azure
+  fi
+  if [[ -n ${AZURE_EMULATOR_ENABLED:-""} ]]; then
+    if [[ -z ${AZURE_STORAGE_API_ENDPOINT:-""} ]] || [[ -z ${AZURITE_HOST:-""} ]] || [[ -z ${AZURE_STORAGE_CONNECTION_STRING:-""} ]]; then
+        usage_azure
+    fi
+    make deploy-azurite
+  else
+    echo "AZURE_EMULATOR_ENABLED is not set. Using real Azure infra for testing."
+  fi
+  setup_azcli
+}
+
+build_and_load() {
+    echo "Building the image and loading it into the kind cluster..."
+    if ! docker image ls | grep -E "^\s*${ETCDBR_IMAGE}\s+${ETCDBR_VERSION}\s+"; then
+      docker build -t ${ETCDBR_IMAGE}:${ETCDBR_VERSION} -f build/Dockerfile .
+    fi
+    kind load docker-image ${ETCDBR_IMAGE}:${ETCDBR_VERSION} --name etcdbr-e2e
+    echo "Successfully loaded the image into the kind cluster."
+}
+
+function create_containers() {
+  for p in ${1//,/ }; do
+    case $p in
+      aws) create_aws_container;;
+      gcp) create_gcp_container;;
+      azure) create_azure_container;;
+      *) echo "Provider: $p is not supported" ;;
+      esac
+  done
+}
+
+function delete_containers() {
+  for p in ${1//,/ }; do
+    case $p in
+      aws) cleanup_aws_container;;
+      gcp) cleanup_gcp_container;;
+      azure) cleanup_azure_container;;
+      *) echo "Provider: $p is not supported" ;;
+      esac
+  done
+}
+
+function cleanup() {
+  if containsElement $STEPS "cleanup" || [ $cleanup_required = true ]; then
+    kind delete cluster --name etcdbr-e2e
+    echo "Cleaning up completed."
```

Comment:
Could the kubeconfig file that is auto generated through kind, also be removed in the cleanup?
If you think that's unnecessary then we don't need to do it.